File: rb/features.rb
# BluefinTecsUserBackoffice SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/audit_feature'
require_relative 'feature/clienttrack_feature'
require_relative 'feature/debug_feature'
require_relative 'feature/idempotency_feature'
require_relative 'feature/log_feature'
require_relative 'feature/metrics_feature'
require_relative 'feature/paging_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/telemetry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module BluefinTecsUserBackofficeFeatures
  def self.make_feature(name)
    case name
    when "base"
      BluefinTecsUserBackofficeBaseFeature.new
    when "audit"
      BluefinTecsUserBackofficeAuditFeature.new
    when "clienttrack"
      BluefinTecsUserBackofficeClienttrackFeature.new
    when "debug"
      BluefinTecsUserBackofficeDebugFeature.new
    when "idempotency"
      BluefinTecsUserBackofficeIdempotencyFeature.new
    when "log"
      BluefinTecsUserBackofficeLogFeature.new
    when "metrics"
      BluefinTecsUserBackofficeMetricsFeature.new
    when "paging"
      BluefinTecsUserBackofficePagingFeature.new
    when "ratelimit"
      BluefinTecsUserBackofficeRatelimitFeature.new
    when "retry"
      BluefinTecsUserBackofficeRetryFeature.new
    when "telemetry"
      BluefinTecsUserBackofficeTelemetryFeature.new
    when "test"
      BluefinTecsUserBackofficeTestFeature.new
    when "timeout"
      BluefinTecsUserBackofficeTimeoutFeature.new
    else
      BluefinTecsUserBackofficeBaseFeature.new
    end
  end
end

File: lua/features.lua
-- BluefinTecsUserBackoffice SDK feature factory

local BaseFeature = require("feature.base_feature")
local AuditFeature = require("feature.audit_feature")
local ClienttrackFeature = require("feature.clienttrack_feature")
local DebugFeature = require("feature.debug_feature")
local IdempotencyFeature = require("feature.idempotency_feature")
local LogFeature = require("feature.log_feature")
local MetricsFeature = require("feature.metrics_feature")
local PagingFeature = require("feature.paging_feature")
local RatelimitFeature = require("feature.ratelimit_feature")
local RetryFeature = require("feature.retry_feature")
local TelemetryFeature = require("feature.telemetry_feature")
local TestFeature = require("feature.test_feature")
local TimeoutFeature = require("feature.timeout_feature")


local features = {}

features.base = function()
  return BaseFeature.new()
end

features["audit"] = function()
  return AuditFeature.new()
end

features["clienttrack"] = function()
  return ClienttrackFeature.new()
end

features["debug"] = function()
  return DebugFeature.new()
end

features["idempotency"] = function()
  return IdempotencyFeature.new()
end

features["log"] = function()
  return LogFeature.new()
end

features["metrics"] = function()
  return MetricsFeature.new()
end

features["paging"] = function()
  return PagingFeature.new()
end

features["ratelimit"] = function()
  return RatelimitFeature.new()
end

features["retry"] = function()
  return RetryFeature.new()
end

features["telemetry"] = function()
  return TelemetryFeature.new()
end

features["test"] = function()
  return TestFeature.new()
end

features["timeout"] = function()
  return TimeoutFeature.new()
end


return features
